lib/src/sdk_env.dart

```diff
@@ -295,18 +295,27 @@ class ToolEnvironment {
   Future<Map<String, dynamic>> _getFlutterVersion() async {
     final result = await runConstrained(
       [..._flutterSdk.flutterCmd, '--version', '--machine'],
+      environment: _flutterSdk.environment,
       throwOnError: true,
     );
-    final waitingForString = 'Waiting for another flutter';
-    return result.parseJson(transform: (content) {
+    return result.parseJson(transform: (String content) {
+      // filter for concurrent flutter execution
+      final waitingForString = 'Waiting for another flutter';
       if (content.contains(waitingForString)) {
-        return content
+        content = content
             .split('\n')
             .where((e) => !e.contains(waitingForString))
             .join('\n');
-      } else {
-        return content;
       }
+      // filter for welcome screen
+      if (content.contains('Welcome to Flutter!')) {
+        final lines = content.split('\n');
+        final separator = lines.indexWhere((l) => l.trim().isEmpty);
+        if (separator >= 0) {
```

Comment:
This looks ok to me, but we almost need a test for this logic - otherwise it might end up doing more harm than good.

Reply:
Added a test for it. Although, I think this is the `unknown unknown` category, where we can't possibly know what will break us, only hope that we can catch it early.

Reply:
True - but I have made "unknown unknown" fixes before, that turned out to themselves be broken, and thus just make things worse :) .